### PR TITLE
Use modern bool comparator on lua

### PR DESCRIPTION
**Pull request by ramon-bernardo**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
- Use modern bool comparator on lua

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

**Review by MillhioreBT — APPROVED** [review]

File: data/scripts/actions/others/flower_pot.lua [hunk 1]
```diff
@@ -54,14 +54,14 @@ local flowerPot = Action()
...
+			if (not flower.watered and not flower.advance) then
 				player:say(flower.msg, TALKTYPE_MONSTER_SAY)
-			elseif (flower.watered == true and flower.advance == false) then
+			elseif (flower.watered and not flower.advance) then
 				target:transform(flower.after)
 				player:say(flower.msg, TALKTYPE_MONSTER_SAY)
 				toPosition:sendMagicEffect(CONST_ME_LOSEENERGY)
 				target:decay()
-			elseif (flower.watered == true and flower.advance == true) then
+			elseif (flower.watered and flower.advance) then
```
Comment (ranisalt) [inline data/scripts/actions/others/flower_pot.lua:64]: While at that, I believe none of these need the parenthesis

Comment (ramon-bernardo) [reply data/scripts/actions/others/flower_pot.lua]: Thanks!

**Review by MillhioreBT — DISMISSED** [review]

**Review by ranisalt — APPROVED** [review]

**Comment by ramon-bernardo** [comment]
Wait, im sending a new commit

**Comment by ramon-bernardo** [comment]
Sorry, i forgot the nil values

File: data/lib/core/achievements.lua [hunk 2]
```diff
@@ -650,7 +650,7 @@ end
 
 function isAchievementSecret(ach)
 	local achievement
-	if tonumber(ach) ~= nil then
```
Comment (ranisalt) [inline data/lib/core/achievements.lua:653]: Are you sure this is similar? Wouldn't a return value of 0 also be falsy in this case?

**Review by ranisalt — CHANGES_REQUESTED** [review]

File: data/lib/core/achievements.lua (hunk 2 above)
Comment (ramon-bernardo) [reply data/lib/core/achievements.lua:653]: No, with [testes](http://tpcg.io/0YHQAG), 0 != nil in this case

File: data/scripts/actions/others/flower_pot.lua [hunk 3]
```diff
@@ -54,16 +54,16 @@ local flowerPot = Action()
...
+			elseif flower.watered and not flower.advance then
 				target:transform(flower.after)
 				player:say(flower.msg, TALKTYPE_MONSTER_SAY)
 				toPosition:sendMagicEffect(CONST_ME_LOSEENERGY)
 				target:decay()
-			elseif (flower.watered and flower.advance) then
+			elseif flower.watered and flower.advance then
 				local i = 1
-				if (math.random(100) <= flower.chance) then
+				if flower.chance >= math.random(100) then
```
Comment (luanluciano93) [inline data/scripts/actions/others/flower_pot.lua:66]: >?

Comment (ramon-bernardo) [reply data/scripts/actions/others/flower_pot.lua:66]: Bad habit, it was just to remove `()`

File: data/lib/core/achievements.lua (hunk 2 above)
Comment (ranisalt) [reply data/lib/core/achievements.lua:653]: Uh, very odd, Lua is the only language I know that does not consider 0 to be falsy...

**Review by ranisalt — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**